packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,261 @@
+  }
+
+  checkFileQueueDuplicity(file: File): boolean {
+    return this.fileQueue.has(this.getUpdatedFileName(file.name));
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getAttachmentElement(file: File, id: string): HTMLElement | null {
+    const dataContainer = new DataTransfer();
+    const attachmentInputElement = document.createElement('input');
+    const snippetTemplate = SelectorEngine.findOne(TEMPLATE_ELEMENT_SELECTOR, this.element) as HTMLTemplateElement;
+    const snippet = snippetTemplate?.content.cloneNode(true) as HTMLElement;
+
+    if (!snippet) {
+      return null;
+    }
+
+    attachmentInputElement.setAttribute('type', 'file');
+    attachmentInputElement.setAttribute('name', `${this.inputName}[]`);
+    attachmentInputElement.setAttribute('hidden', '');
+    dataContainer.items.add(file);
+    attachmentInputElement.files = dataContainer.files;
+
+    const item = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="item"]`);
+    const itemName = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="name"]`);
+    const itemButton = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="button"]`);
+
+    item!.appendChild(attachmentInputElement);
+    item!.setAttribute('id', id);
+    itemName!.innerHTML = file.name;
+    itemButton!.setAttribute(DATA_DISMISS_ATTRIBUTE, id);
+    item!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+    itemName!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+    itemButton!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+
+    return snippet;
+  }
+
+  createAttachmentEvents(id: string) {
+    const currentItem = SelectorEngine.findOne(`[${DATA_DISMISS_ATTRIBUTE}="${id}"]`);
+    currentItem &&
+      EventHandler.on(currentItem, 'click', (event: Event) => {
+        event.preventDefault();
+        const target = (event.target as HTMLElement)?.dataset?.dismiss;
+        if (target) {
+          this.removeFromQueue(target);
+        }
+      });
+  }
+
+  appendToList(file: File) {
+    if (!this.listElement) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Unfortunately, there is no list element created for inserting files to upload ⚠️');
+      }
+
+      return;
+    }
+
+    if (!this.isMultiple) {
+      this.clearFileQueue();
+    }
+
+    const id = this.getUpdatedFileName(file.name);
+    const attachment = this.getAttachmentElement(file, id);
+
+    if (!attachment) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Cannot find element with template to render ⚠️');
+      }
+
+      return;
+    }
+
+    // Appending an item into queue
+    this.fileQueue.set(id, file);
+
+    // Appending node into list
+    this.listElement.appendChild(attachment);
+
+    // Creating an event for the delete item button
+    this.createAttachmentEvents(id);
+
+    // Reset drag state
+    this.dragReset();
+  }
+
+  addToQueue(file: File) {
+    const checkedFileSize = this.checkAllowedFileSize(file);
```

Comment:
```suggestion
    const isFileSizeAllowed = this.isFileSizeAllowed(file);
```

I suggest for better readability renaming at least the variable because from `checkAllowedFileSize` I do not read for which state I get true/false. E. g. if the name is `isFileSizeAllowed()` I know at first sight that when the file is bigger I get false and the opposite. Same for duplicity -> `hasFileQueueDuplicity`, `isFileQueueUnique`?

Reply:
done